examples/templates/winapi_sockets.h

```diff
@@ -0,0 +1,79 @@
+#ifndef __WINAPI_SOCKET_TEMPLATE_H__
```

Comment:
I want to see if I can merge this with the posix one, is that ok with you @LiamBindle ?

Reply:
Again, I'd like to keep them seperate. In fact, I'd be happy to have a winsockets example, but I want to keep the examples as basic unix examples.

Reply:
Ok, then I'll keep the winapi stuff separated.